c7n/resources/rds.py

```diff
@@ -912,6 +918,160 @@ def process_resource_set(self, snaps, tag_keys):
+        the rds api docs for the method. Specifically
+        DestinationRegion is a new parameter that's not current in the
+        json api models. Additionally TargetDBSnapshotIdentifier is
+        not a requirement for the request.
+
+        We do the legwork nesc to work around.
+        """
+        # https://goo.gl/dPaX2L
+        params = dict(
+            DestinationRegion=self.data['target_region'],
+            SourceDBSnapshotIdentifier=snapshot['DBSnapshotArn'],
+            #TargetDBSnapshotIdentifier=snapshot['DBSnapshotIdentifier'] + '22',
+            KmsKeyId=key)
+        op = self.get_presign_operation(source)
+
+        request_dict = source._serializer.serialize_to_request(
+            params, op)
+        #request_dict['body']['SignatureMethod'] = 'HmacSHA256'
+        #request_dict['body']['SignatureVersion'] = '4'
+        request_dict['headers']['X-Amz-Content-SHA256'] = EMPTY_SHA256_HASH
+        prepare_request_dict(
+            request_dict, endpoint_url=source.meta.endpoint_url)
+        #print request_dict
+        return source._request_signer.generate_presigned_url(
+            request_dict=request_dict, expires_in=86400,
+            operation_name='CopyDBSnapshot')
+
+    def process_resource(self, target, source, key, tags, snapshot):
+        p = {}
+
+        if "KmsKeyId" in snapshot:
+            p['PreSignedUrl'] = self.get_presigned_url(
+                source, target, key, snapshot)
+            # Try and verify why the copydbsnapshot api doesn't like our
+            # valid pre-signed url.
+
+            #import urlparse, pprint
+            #parsed = urlparse.urlparse(p['PreSignedUrl'])
+            #d = urlparse.parse_qs(parsed.query)
+            #pprint.pprint(d)
+            #import requests
+            #response = requests.post(p['PreSignedUrl'])
+            #print response
+            #print response.text
+
+        if key:
+            p['KmsKeyId'] = key
+
+        p['TargetDBSnapshotIdentifier'] = snapshot[
+            'DBSnapshotIdentifier'].replace(':', '-')
+        p['SourceRegion'] = self.manager.config.region
+        p['SourceDBSnapshotIdentifier'] = snapshot['DBSnapshotArn']
+
+        if self.data.get('copy_tags', True):
+            p['CopyTags'] = True
+        if tags:
+            p['Tags'] = tags
+        result = target.copy_db_snapshot(**p)
+        snapshot['c7n:CopiedSnapshot'] = result[
+            'DBSnapshot']['DBSnapshotIdentifier']
+
+    def process_resource_set(self, resource_set):
+        target_client = self.manager.session_factory(
+            region=self.data['target_region']).client('rds')
+        target_key = self.data.get('target_key')
+        tags = [{'Key': k, 'Value': v} for k, v in self.data.get('tags', {})]
+        source_client = local_session(
+            self.manager.session_factory).client('rds')
+
+        for snapshot_set in chunks(resource_set, 5):
+            for r in snapshot_set:
+                self.process_resource(
```

Comment:
this needs to catch on hitting the 5 in progress limit and await. we need to mark the snapshot transfer actions as not suitable for lambda due to need to coordinate around the limit.